test/subtype.jl

```diff
@@ -136,6 +136,35 @@ function test_diagonal()
     @test  issub(Tuple{Tuple{T, T}} where T>:Int, Tuple{Tuple{T, T} where T>:Int})
     @test  issub(Vector{Tuple{T, T} where Number<:T<:Number},
                  Vector{Tuple{Number, Number}})
+
+    @test !issub(Type{Tuple{T,Any} where T},   Type{Tuple{T,T}} where T)
```

Comment:
I'm confused...
```julia
julia> Type{Tuple{T, Any} where T} <: Type{Tuple{Any,Any}} <: Type{Tuple{T, T}} where T
true

julia> Type{Tuple{T, Any} where T} <:                         Type{Tuple{T, T}} where T
false
```
In fact, we have
```julia
julia> Type{Tuple{T, Any} where T} == Type{Tuple{Any,Any}}
true
```
which looks right to me, as `(Tuple{T, Any} where T) == Tuple{Any,Any}`, noting the covariance of `T` here. But `Type{Tuple{Any,Any}} <: Type{Tuple{T, T}} where T` also looks fine: LHS and RHS become equal for one specific choice of `T` on the RHS, namely `Any`. And `T` does not fall under the diagonal rule. (It would if the `where` were inside the `Type{}`.) So that would leave me with the conclusion that `Type{Tuple{T, Any} where T} <: Type{Tuple{T, T}} where T` should hold. Yet here, we test for the opposite. Wrong test or wrong reasoning? (Note that as it is now, this violates transitivity, so there must be a bug somewhere.)

Reply:
The same violation of transitivity can be observed for this simpler case, BTW:
```julia
julia> Ref{Tuple{T} where T} <: Ref{Tuple{Any}} <: Ref{Tuple{T}} where T
true

julia> Ref{Tuple{T} where T} <:                    Ref{Tuple{T}} where T
false
```
I'd guess we have an issue for this somewhere, but couldn't find one at a quick glance.